src/olympia/files/management/commands/extract_host_permissions.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+from django.core.management.base import BaseCommand
+
+import olympia.core.logger
+from olympia import amo
+from olympia.amo.celery import create_chunked_tasks_signatures
+from olympia.files.models import File
+from olympia.files.tasks import extract_host_permissions
+
+
+class Command(BaseCommand):
+    help = 'Extract host permissions from manifests in stored xpis.'
+
+    def handle(self, *args, **options):
+        log = olympia.core.logger.getLogger('z.files')
+        files = File.objects.filter(version__addon__type=amo.ADDON_EXTENSION).order_by(
+            'pk'
+        )
```

Comment:
Not sure we particularly need to extract permissions from files that are already disabled but I guess it doesn't hurt.

Reply:
It's basically the same code we ran when populating `optional_permissions`, so I didn't really consider that.